src/hardware/mouse/mouse_interfaces.cpp

```diff
@@ -0,0 +1,818 @@
+			break;
+		case MouseInterfaceId::COM4:
+			mouse_interfaces.push_back(&interface_com4);
+			break;
+		default: assert(false); break;
+		}
+
+	for (auto interface : mouse_interfaces)
+		interface->Init();
+}
+
+MouseInterface *MouseInterface::Get(const MouseInterfaceId interface_id)
+{
+	const auto idx = static_cast<size_t>(interface_id);
+	if (idx < mouse_interfaces.size())
+		return mouse_interfaces[idx];
+
+	assert(interface_id == MouseInterfaceId::None);
+	return nullptr;
+}
+
+MouseInterface *MouseInterface::GetDOS()
+{
+	const auto idx = static_cast<uint8_t>(MouseInterfaceId::DOS);
+	return MouseInterface::Get(static_cast<MouseInterfaceId>(idx));
+}
+
+MouseInterface *MouseInterface::GetPS2()
+{
+	const auto idx = static_cast<uint8_t>(MouseInterfaceId::PS2);
+	return MouseInterface::Get(static_cast<MouseInterfaceId>(idx));
+}
+
+MouseInterface *MouseInterface::GetSerial(const uint8_t port_id)
+{
+	if (port_id < SERIAL_MAX_PORTS) {
+		const auto idx = static_cast<uint8_t>(MouseInterfaceId::COM1) + port_id;
+		return MouseInterface::Get(static_cast<MouseInterfaceId>(idx));
+	}
+
+	LOG_ERR("MOUSE: Ports above COM4 not supported");
+	assert(false);
+	return nullptr;
+}
+
+MouseInterface::MouseInterface(const MouseInterfaceId interface_id,
+                               const float sensitivity_predefined)
+        : interface_id(interface_id),
+          sensitivity_predefined(sensitivity_predefined)
+{
+	mouse_info.interfaces.emplace_back(MouseInterfaceInfoEntry(interface_id));
+}
+
+void MouseInterface::Init()
+{
+	// At this point configuration should already be loaded,
+	// so the default sensitivity is known
+	ConfigResetSensitivity();
+}
+
+uint8_t MouseInterface::GetInterfaceIdx() const
+{
+	return static_cast<uint8_t>(interface_id);
+}
+
+bool MouseInterface::IsMapped() const
+{
+	return mapped_idx < mouse_info.physical.size();
+}
+
+bool MouseInterface::IsMapped(const uint8_t device_idx) const
+{
+	return mapped_idx == device_idx;
+}
+
+bool MouseInterface::IsEmulated() const
+{
+	return emulated;
+}
+
+bool MouseInterface::IsUsingEvents() const
+{
+	return IsEmulated() && (map_status == MouseMapStatus::HostPointer ||
+	                        map_status == MouseMapStatus::Mapped);
+}
+
+bool MouseInterface::IsUsingHostPointer() const
+{
+	return IsEmulated() && (map_status == MouseMapStatus::HostPointer);
+}
+
+uint16_t MouseInterface::GetMinRate() const
+{
+	return min_rate_hz;
+}
+
+MouseInterfaceId MouseInterface::GetInterfaceId() const
+{
+	return interface_id;
+}
+
+MouseMapStatus MouseInterface::GetMapStatus() const
+{
+	return map_status;
+}
+
+uint8_t MouseInterface::GetMappedDeviceIdx() const
+{
+	return mapped_idx;
+}
+
+int16_t MouseInterface::GetSensitivityX() const
+{
+	return sensitivity_user_x;
+}
+
+int16_t MouseInterface::GetSensitivityY() const
+{
+	return sensitivity_user_y;
+}
+
+uint16_t MouseInterface::GetRate() const
+{
+	return rate_hz;
+}
+
+void MouseInterface::NotifyInterfaceRate(const uint16_t rate_hz)
```

Comment:
`rate_hz` is also a member of this `MouseInterface` class. 
Suggest a new argument name to avoid ambiguity/shadowing in the `interface_rate_hz = rate_hz` statement.

Reply:
OK, changed argument name.